### PR TITLE
Increase timeout of Windows test job

### PR DESCRIPTION
**Pull request by dougbu**
- no consistency in failures, maybe we have more tests now?
  - both of the runs below involved test failures but results were not uploaded
- https://dev.azure.com/dnceng/public/_build/results?buildId=768850
- https://dev.azure.com/dnceng/public/_build/results?buildId=768703

### CODE REVIEW

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -517,6 +517,7 @@ stages:
         jobDisplayName: "Test: Windows Server 2016 x64"
         agentOs: Windows
         isTestingJob: true
+        timeoutInMinutes: 240
```
Comment (wtgodbe) [inline .azure/pipelines/ci.yml:520]: Up from 180 or 120?

**Review by wtgodbe — APPROVED** [review]

File: .azure/pipelines/ci.yml (hunk 1 above)
Comment (dougbu) [reply .azure/pipelines/ci.yml:520]: 180

**Comment by dougbu** [comment]
Surprised to see the job that had been timing out succeeded in only 49 minutes because the two failing builds this morning logged test failures as well as the timeout. Will look at the `*.dmp` files from the failures before my final decision on this PR.

**Comment by dougbu** [comment]
I'm not seeing any test assemblies in the `*.dmp` files. The smaller dump for https://dev.azure.com/dnceng/public/_build/results?buildId=768850 shows code in VBCSCompiler.dll was running while that for https://dev.azure.com/dnceng/public/_build/results?buildId=768703 was running Microsoft.Build.dll code (with the compiler nowhere in sight).

@HaoK @BrennanConroy and @davidfowl will dumps be collected just because a process happens to be running when a build is cancelled❔ Alternatively is there a way to determine what the `dotnet` process is _really_ waiting for❔

**Comment by dougbu** [comment]
Note the most recent rolling build was successful but didn't include any changes except to doc comments: https://github.com/dotnet/aspnetcore/compare/fa2d86e5b6b88dd8d7ec4b6de9ca5640e64ed76f...9f0ae10cabde4ae2e62e13a2d1b234c5de42a67f

**Comment by HaoK** [comment]
The new stuff for hangs and dumps was helix specific, so its not going to help for windows test jobs unfortunately

**Comment by dougbu** [comment]
:ok: Since the job usually finishes well before its current 180 minute timeout and just uploading a `*.dmp` file can take 15 minutes, I'm going to increase (double) the `cancelTimeoutInMinutes` instead. Hope is that will give job enough time to upload test results as well as any dumps.

**Comment by dougbu** [comment]
I meant artifacts/logs/

**Comment by dougbu** [comment]
@MattGal @ilyas1974 is something odd happening w/ https://dev.azure.com/dnceng/public/_build/results?buildId=769815❔ The job w/ the increased cancel timeout has been sitting in Queued state for a couple of hours now. Later builds seem to be getting Windows.Server.Amd64.VS2019.Open machines ahead of it.

**Comment by dougbu** [comment]
More to the point, is my best option cancelling that build and starting the pipeline again❔

**Comment by ilyas1974** [comment]
it appears that the build is currently in progress so I do not believe any additional action needs to be taken.

**Comment by dougbu** [comment]
Weird. I'm glad the job finally started running❕

**Comment by MattGal** [comment]
> More to the point, is my best option cancelling that build and starting the pipeline again❔

If it's been > 90 minutes, definitely so.  In those scenarios you're hitting https://github.com/dotnet/core-eng/issues/10136; in this situation we just don't have enough machines and AzDO has a hard 90 minute timeout they've offered to eventually extend, but who knows when...

**Pull request merged**